### ISSUE TITLE
CPUAccelerator devices should be an int > 0

### ISSUE THREAD

**Issue by tempoche1** [Noble-Lab/casanovo#109]
Hi,

I am following the Casanovo getting started page, and running `casanovo --mode=denovo --peak_path=[PATH_TO]/sample_preprocessed_spectra.mgf` on the small example MGF file results in an error on my computer.

I use a Macbook Air, M2 chip, macOS 12.5.1.

I was able to create a conda environment and install Casanovo with no errors:
- `conda create --name casanovo_env python=3.10`
- `conda activate casanovo_env`
- `pip install casanovo`

Running `casanovo --help` actually produced an error: `ModuleNotFoundError: No module named 'tensorboard'` (full error msg: [help_error_msg.txt](https://github.com/Noble-Lab/casanovo/files/10335321/help_error_msg.txt)). However the --help page ran correctly after I installed tensorboard (by `conda install tensorboard`) .

The problem occurred when I ran `casanovo --mode=denovo --peak_path=data/casanovo_demo/sample_preprocessed_spectra.mgf`, producing the TypeError: \`devices\` selected with \`CPUAccelerator\` should be an int > 0 (full error msg: [run_error_msg.txt](https://github.com/Noble-Lab/casanovo/files/10335370/run_error_msg.txt)).

I looked into the source code and error message, and noticed that running `trainer = pl.Trainer(devices=-1)` (after `import pytorch_lightning as pl`) produces the same final TypeError message as above. I also noticed on my computer that the function `_get_devices()` from casanovo/denovo/model_runner.py returns -1. My guess is this is the core problem, although I also noticed a note in the `_get_strategy()` function mentioning failure on Apple Silicon chips, so I'm not sure.

Edit: Also, here is the casanovo log file from that run [casanovo_20230103195034.log](https://github.com/Noble-Lab/casanovo/files/10335736/casanovo_20230103195034.log)

Any help would be appreciated,
Temana

**Comment by bittremieux** [comment]
I guess the problem is that with the MPS backend it's not possible to specify `-1` as the number of devices. The fix would be to [remove `"backend.mps"` here](https://github.com/Noble-Lab/casanovo/blob/main/casanovo/denovo/model_runner.py#L379), and go to one of the other statements in that if-else on Mac. I don't have a Mac, so @wfondrie can you try it and confirm this fixes the issue?

**Comment by temana-short** [comment]
I'm not too familiar with editing/running python packages, but I just tried the above suggestion and it did fix the error, producing the .mztab output with PSMs on the provided small example.

However, the process did take a while (2 mins, longer than the "< 1 minute" suggested on the documentation for the small .mgf file), and the log file had 3 WARNINGs relating to pytorch, at least one of which was related to performance.

I've attached the exact command-line runs: [cmd_line_runs.txt](https://github.com/Noble-Lab/casanovo/files/10926636/cmd_line_runs.txt) and log file from the successful casanovo run: [casanovo_20230309115235.log](https://github.com/Noble-Lab/casanovo/files/10926637/casanovo_20230309115235.log).

Any help would be appreciated,
Temana
